src/main/resources/templates/History.html

```diff
@@ -0,0 +1,340 @@
+        input.placeholder = "Filter by " + column;
+        filterInput.appendChild(input);
+        filterSection.appendChild(filterInput);
+    });
+
+    function filterTable(columnIndex) {
+        var input, filter, table, tr, td, i, txtValue;
+        input = document.getElementById(columns[columnIndex].replace(/\s+/g, '') + "Filter");
+        filter = input.value.toUpperCase();
+
+        // Split the filter into include and exclude parts
+        var filters = filter.split(';').map(f => f.trim());
+        var includeFilters = filters.filter(f => !f.startsWith('~'));
+        var excludeFilters = filters.filter(f => f.startsWith('~')).map(f => f.substring(1));
+
+        table = document.getElementById("feedbackTable");
+        tr = table.getElementsByTagName("tr");
+        for (i = 0; i < tr.length; i++) {
+            td = tr[i].getElementsByTagName("td")[columnIndex];
+            if (td) {
+                txtValue = td.textContent || td.innerText;
+                var matchesInclude = includeFilters.length === 0 || includeFilters.some(filter => txtValue.toUpperCase().includes(filter));
+                var matchesExclude = excludeFilters.some(filter => txtValue.toUpperCase().includes(filter));
+
+                // Show the row if it matches any of the inclusion filters and none of the exclusion filters
+                if (matchesInclude && !matchesExclude) {
+                    tr[i].style.display = "";
+                } else {
+                    tr[i].style.display = "none";
+                }
+            }
+        }
+    }
+
+    function exportTableToCSV() {
+        var csv = [];
+        var rows = document.querySelectorAll("#feedbackTable tbody tr");
+
+        // Filter and sort the rows based on the current sorting criteria and filters
+        var filteredRows = Array.from(rows).sort(compareRows);
+
+        // Loop through each row and collect cell data
+        filteredRows.forEach(function(row) {
+            if (row.style.display !== "none") {
+                var rowData = [];
+                row.querySelectorAll("td").forEach(function (cell) {
+                    rowData.push(cell.textContent.trim());
+                });
+                csv.push(rowData.join(","));
+            }
+        });
+
+        // Convert the CSV array to a string
+        var csvString = csv.join("\n");
+
+        // Create a Blob object to save the CSV data
+        var blob = new Blob([csvString], { type: "text/csv;charset=utf-8;" });
+
+        // Create a temporary link element to trigger the download
+        var link = document.createElement("a");
+        var url = URL.createObjectURL(blob);
+        link.setAttribute("href", url);
+        link.setAttribute("download", "table.csv");
+        link.style.visibility = "hidden";
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+    }
+
+    // function exportTableToCSV() {
```

Comment:
will we still need this code? why did you leave it commented?

Reply:
oopsie